core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/AkkaActorRefMappingService.scala

```diff
@@ -84,6 +84,15 @@ class AkkaActorRefMappingService(actorService: AkkaActorService) extends AmberLo
     queriedActorVirtualIdentities.clear()
   }
 
+  def findActorVirtualIdentity(ref: ActorRef): Option[ActorVirtualIdentity] = {
+    actorRefMapping
+      .find {
+        case (_, actorRef) =>
+          actorRef == ref
+      }
+      .map(_._1)
+  }
```

Comment:
in what case will it not find a ref and return a None?

Reply:
if an actor is crashed, and its ref is removed from the mapping.